scipy/interpolate/_rgi.py

```diff
@@ -225,12 +228,15 @@ class RegularGridInterpolator:
     # this class is based on code originally programmed by Johannes Buchner,
     # see https://github.com/JohannesBuchner/regulargrid
 
-    _SPLINE_DEGREE_MAP = {"slinear": 1, "cubic": 3, "quintic": 5, 'pchip': 3}
+    _SPLINE_DEGREE_MAP = {"slinear": 1, "cubic": 3, "quintic": 5, 'pchip': 3,
+                          "slinear_": 1, "cubic_": 3, "quintic_": 5,}
+    _SPLINE_METHODS_recurive = {"slinear_", "cubic_", "quintic_", "pchip"}
```

Comment:
I'm not a fan of having the `_` names. Maybe have something a bit more descriptive to show what differentiates them from the existing methods?

Reply:
Could we not have `cubic_legacy` or similar?

Reply:
Thanks for the suggestion, done.